ndcube/utils/wcs.py

```diff
@@ -477,3 +478,35 @@ def compare_wcs_physical_types(source_wcs, target_wcs):
     target_wcs = get_low_level_wcs(target_wcs, 'target_wcs')
 
     return source_wcs.world_axis_physical_types == target_wcs.world_axis_physical_types
+
+
+def identify_invariant_axes(source_wcs, target_wcs, input_shape):
```

Comment:
Do we need to have an `input_shape` here? Could we just use `pixel_n_dim` from the WCS and pass some static number to `np.arange()` below?

Reply:
I think we do need the `input_shape` here.